lib/methodrecord.ts

```diff
@@ -0,0 +1,155 @@
+import { GetIntrinsic } from './intrinsicclass';
+import type { Temporal } from '..';
+import type { CalendarParams } from './internaltypes';
+import type { TimeZonePrototypeKeys } from './intrinsicclass';
+
+// Not all calendar protocol methods need to be able to be cached
+type CalendarRecordMethodNames =
+  | 'dateAdd'
+  | 'dateFromFields'
+  | 'dateUntil'
+  | 'day'
+  | 'fields'
+  | 'mergeFields'
+  | 'monthDayFromFields'
+  | 'yearMonthFromFields';
+
+type CalendarRecordInfo = {
+  Name: 'Calendar';
+  Protocol: Temporal.CalendarProtocol;
+  MethodName: CalendarRecordMethodNames;
+};
+
+type TimeZoneRecordInfo = {
+  Name: 'TimeZone';
+  Protocol: Temporal.TimeZoneProtocol;
+  MethodName: TimeZonePrototypeKeys;
+};
+
+// We switch off type checking when accessing the cached methods on this object.
```

Comment:
Agree, no need to worry about making this work if you're gonna remove them soon.